### ISSUE TITLE
Question: Are there any known issues with the `jax.random.PRNGKey` when running HMC models in Numpyro?

### ISSUE THREAD

**Issue by mdenhollander** [pyro-ppl/numpyro#1036]
Hey all,

I had a question regarding the `jax.random.PRNGKey` function. 

When working with Numpyro I noticed that the parameter samples can slightly vary between model runs, even after setting a fixed value for the random seed through `jax.PRNGKey()`. I am using Numpyro in a Class-based structure, in which the Jax random key is created in the `__init__` function of the class and saved as an attribute, which is then called in every stochastic function (e.g. `numpyro.sample()`). 

The package versions I am using are Jax version 0.2.3, Jaxlib version 0.1.56 and Numpyro version 0.4.1 .

Hence, my question is whether there are any known issues with the random seed in any of these library versions? Or, alternatively, could using a set-up with Python Classes somehow interfere with setting a PRNGKey in Numpyro? But in general, the main reason for this issue is to get to know if there are any known issues with the PRNGKey function of Jax and if there is any way this can be tackled?

Please let me know what your experiences are.
Many thanks in advance!

**Comment by mdenhollander** [comment]
Thanks for your reply!

If that would be the case, is it recommended to disable fast math mode? And any ideas on what the computational efficiency loss could be then?

**Comment by fehiepsi** [comment]
I haven't seen any problem with fast math mode for a while. Since you are using an old jax version, it is hard to tell.

**Comment by fehiepsi** [comment]
Hi @mdenhollander, please feel free to reopen this topic again if you think we have some issues with HMC implementation. We typically use github for issues and feature requests. For questions and discussions, we recommend using the [forum](https://forum.pyro.ai/). :)

**Comment by mdenhollander** [comment]
Thanks for your help! For any future questions, I will definitely use the Forum
